cosmos/providers/dbt/core/operators.py

```diff
@@ -18,73 +19,81 @@
 )
 
 
+class DbtArgs(TypedDict, total=False):
+    """
+    Maintain all available arguments here so that we can help end users with available kwargs.
+    """
+
+    project_dir: Path  # Not available for end users, generated by Cosmos
+    base_cmd: str | list[str]
+    select: str
+    exclude: str
+    selector: str
+    vars: dict[str, str]
+    models: str  # Not available for end users, generated by Cosmos
+    cache_selected_only: bool
+    no_version_check: bool
+    fail_fast: bool
+    quiet: bool
+    warn_error: bool
+    db_name: str
+    schema: str
+    env: dict[str, str]
+    append_env: bool
+    output_encoding: str
+    skip_exit_code: int
+    cancel_query_on_kill: bool
+    dbt_executable_path: str
+    full_refresh: bool  # Available on seed only
+    macro_name: str  # Available on run operation only
+
+
```

Comment:
we're trying to maintain python 3.7 support (which admittedly we should have CI/CD set up for, but we haven't gotten around to it yet). this makes typing a bit tougher - the things I know to look out for:
- no unions via pipes, need to use `typing.Union`
- no `list` or `dict`, need to use `typing.List` and `typing.Dict`

Do you mind making those changes? I've suggested some stuff in-line here, but it may not be 100% correct
```suggestion
class DbtArgs(TypedDict, total=False):
    """
    Maintain all available arguments here so that we can help end users with available kwargs.
    """

    project_dir: Path  # Not available for end users, generated by Cosmos
    base_cmd: Union[str, List[str]]
    select: str
    exclude: str
    selector: str
    vars: Dict[str, str]
    models: str  # Not available for end users, generated by Cosmos
    cache_selected_only: bool
    no_version_check: bool
    fail_fast: bool
    quiet: bool
    warn_error: bool
    db_name: str
    schema: str
    env: Dict[str, str]
    append_env: bool
    output_encoding: str
    skip_exit_code: int
    cancel_query_on_kill: bool
    dbt_executable_path: str
    full_refresh: bool  # Available on seed only
    macro_name: str  # Available on run operation only


```

Reply:
The `from __future__ import annotations` import for [PEP 563](https://peps.python.org/pep-0563/) use should handle these typing concerns though. For example, Airflow itself also supports 3.7 and uses PEP 563. No issues thus far. However, `TypedDict` will require some sort of Python version check since it wasn't part of `typing` until Python 3.8. Airflow [uses a `typing_compat` module](https://github.com/apache/airflow/blob/47b67f13da667de4eb69233404d4d5a33574dd58/airflow/typing_compat.py#L32) for this.

Reply:
Yep, there are already a handful of places in the main code base making use of this import, that's where I got the inspiration from.
https://github.com/astronomer/astronomer-cosmos/blob/main/cosmos/providers/dbt/core/operators.py#L89

I didn't keep in mind that TypedDict didn't exist in 3.7 though so I'll make use of how Airflow handles it.